tests/unit/test-class-wc-payments-express-checkout-button-display-handler.php

```diff
@@ -0,0 +1,286 @@
+		$this->mock_wcpay_account = $this->createMock( WC_Payments_Account::class );
+
+		$this->mock_wcpay_gateway = $this->make_wcpay_gateway();
+
+		$this->mock_platform_checkout_utilities = $this->getMockBuilder( Platform_Checkout_Utilities::class )
+			->disableOriginalConstructor()
+			->setMethods( [ 'is_country_available' ] )
+			->getMock();
+
+		$this->mock_woopay_button_handler = $this->getMockBuilder( WC_Payments_Platform_Checkout_Button_Handler::class )
+			->setConstructorArgs(
+				[
+					$this->mock_wcpay_account,
+					$this->mock_wcpay_gateway,
+					$this->mock_platform_checkout_utilities,
+				]
+			)
+			->setMethods(
+				[
+					'is_woopay_enabled',
+					'should_show_platform_checkout_button',
+				]
+			)
+			->getMock();
+
+		$this->mock_payment_request_button_handler = $this->getMockBuilder( WC_Payments_Payment_Request_Button_Handler::class )
+			->setConstructorArgs(
+				[
+					$this->mock_wcpay_account,
+					$this->mock_wcpay_gateway,
+				]
+			)
+			->setMethods(
+				[
+					'should_show_payment_request_button',
+				]
+			)
+			->getMock();
+
+			$this->express_checkout_button_display_handler = new WC_Payments_Express_Checkout_Button_Display_Handler( $this->mock_wcpay_gateway, $this->mock_payment_request_button_handler, $this->mock_woopay_button_handler );
+
+		add_filter(
+			'woocommerce_available_payment_gateways',
+			function() {
+				return [ 'woocommerce_payments' => $this->mock_wcpay_gateway ];
+			}
+		);
+	}
+
+	/**
+	 * @return WC_Payment_Gateway_WCPay
+	 */
+	private function make_wcpay_gateway() {
+		$mock_customer_service         = $this->createMock( WC_Payments_Customer_Service::class );
+		$mock_token_service            = $this->createMock( WC_Payments_Token_Service::class );
+		$mock_action_scheduler_service = $this->createMock( WC_Payments_Action_Scheduler_Service::class );
+		$mock_rate_limiter             = $this->createMock( Session_Rate_Limiter::class );
+		$mock_order_service            = $this->createMock( WC_Payments_Order_Service::class );
+
+		return new WC_Payment_Gateway_WCPay(
+			$this->mock_api_client,
+			$this->mock_wcpay_account,
+			$mock_customer_service,
+			$mock_token_service,
+			$mock_action_scheduler_service,
+			$mock_rate_limiter,
+			$mock_order_service
+		);
+	}
+
+	public function test_display_express_checkout_separator_if_necessary_all_enabled() {
+		$this->mock_woopay_button_handler
+			->method( 'is_woopay_enabled' )
+			->willReturn( true );
+
+		$this->mock_woopay_button_handler
+			->method( 'should_show_platform_checkout_button' )
+			->willReturn( true );
+
+		$this->mock_payment_request_button_handler
+			->method( 'should_show_payment_request_button' )
+			->willReturn( true );
+
+		$this->express_checkout_button_display_handler->display_express_checkout_separator_if_necessary();
+
+		$this->assertStringContainsString( 'wcpay-payment-request-button-separator', ob_get_contents() );
```

Comment:
```suggestion
		ob_start();
		$this->assertStringContainsString( 'wcpay-payment-request-button-separator', ob_get_contents() );
		ob_end_clean();
```
Just a heads up, I think you need to manually open, close, and clean this buffer to ensure other tests are not affected. This applies for everywhere else where you've used `ob_get_contents()` as well. Let me know if I'm misunderstanding anything though!

Reply:
I do believe you're correct. Thanks for pointing it out!